tests/components/netatmo/test_climate.py

```diff
@@ -0,0 +1,854 @@
+    ATTR_PRESET_MODE,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    PRESET_AWAY,
+    PRESET_BOOST,
+)
+from homeassistant.components.netatmo import climate
+from homeassistant.components.netatmo.climate import (
+    NA_THERM,
+    NA_VALVE,
+    PRESET_FROST_GUARD,
+    PRESET_SCHEDULE,
+)
+from homeassistant.components.netatmo.const import (
+    ATTR_SCHEDULE_NAME,
+    SERVICE_SET_SCHEDULE,
+)
+from homeassistant.const import ATTR_ENTITY_ID, ATTR_TEMPERATURE
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+
+async def test_setup_no_data(hass, entry_error):
+    """Test setup with erroneous data."""
+    await hass.async_block_till_done()
+
+    assert (
+        hass.data["netatmo"][entry_error.entry_id]["netatmo_data_handler"].data[
+            "HomeData"
+        ]
+        is None
+    )
+
+
+async def test_webhook_event_handling_thermostats(hass, climate_entry):
+    """Test service and webhook event handling with thermostats."""
+    await hass.async_block_till_done()
+
+    assert (
+        hass.data["netatmo"][climate_entry.entry_id]["netatmo_data_handler"].data[
+            "HomeData"
+        ]
+        is not None
+    )
+
+    climate_entity_livingroom = "climate.netatmo_livingroom"
+
+    assert hass.states.get(climate_entity_livingroom).state == "auto"
+    assert (
+        hass.states.get(climate_entity_livingroom).attributes["preset_mode"]
+        == "Schedule"
+    )
+    assert hass.states.get(climate_entity_livingroom).attributes["temperature"] == 12
+
+    # Test service setting the temperature
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_TEMPERATURE,
+        {ATTR_ENTITY_ID: climate_entity_livingroom, ATTR_TEMPERATURE: 21},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+
+    # Fake webhook thermostat manual set point
+    webhook_data = {
+        "user_id": "91763b24c43d3e344f424e8b",
+        "user": {"id": "91763b24c43d3e344f424e8b", "email": "john@doe.com"},
+        "home_id": "91763b24c43d3e344f424e8b",
+        "room_id": "2746182631",
+        "home": {
+            "id": "91763b24c43d3e344f424e8b",
+            "name": "MYHOME",
+            "country": "DE",
+            "rooms": [
+                {
+                    "id": "2746182631",
+                    "name": "Livingroom",
+                    "type": "livingroom",
+                    "therm_setpoint_mode": "manual",
+                    "therm_setpoint_temperature": 21,
+                    "therm_setpoint_end_time": 1612734552,
+                }
+            ],
+            "modules": [
+                {"id": "12:34:56:00:01:ae", "name": "Livingroom", "type": "NATherm1"}
+            ],
+        },
+        "mode": "manual",
+        "event_type": "set_point",
+        "temperature": 21,
+        "push_type": "display_change",
+    }
+    async_dispatcher_send(
+        hass,
+        "signal-netatmo-webhook-set_point",
```

Comment:
Can we trigger the netatmo webhook handler by calling the webhook integration handler instead?

https://github.com/home-assistant/core/blob/9ec4c07753f12da47dba351bbe2f1156664d6a63/homeassistant/components/webhook/__init__.py#L72